use libuuid instead of zuuid

garlick opened the pull request:
Since zuuid asserts when `/dev/urandom` cannot be opened (as hit recently in #2593), this PR replaces it with `libuuid`, a better option because it implements an internal fallback to pseudo-random when that happens, and also because its functions don't require error checking.

Unfortunately, this adds a new dependency, but one we already required in flux-security.

This fixes #2281

grondo commented on src/test/docker/bionic/Dockerfile:
```diff
@@ -58,6 +58,7 @@ RUN apt-get -qq install -y --no-install-recommends \
         libsodium-dev \
         libzmq3-dev \
         libczmq-dev \
+	uuid-dev \
```
uuid-dev is already listed below.

grondo commented on src/test/docker/centos7/Dockerfile:
```diff
@@ -28,6 +28,7 @@ RUN yum -y update \
       ccache \
       cppcheck \
       czmq-devel \
+      libuuid-devel \
```
uuid-devel, libuuid-devel are already listed below. (Though I do wonder why we currently have both names, seems like one should be invalid.)

garlick commented:
Ah I should have checked before adding the dependencies to the Dockerfiles.  OK, dropped the dockerfile commit, rebased, and forced a push.

I dunno about the centos dockerfile's inclusion of the package by both names...leaving as is for now.

grondo commented:
Thanks! Added merge-when-passing.
